File: 0455-assign-cookies/0455-assign-cookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    g.sort((a, b) => a - b)
    s.sort((a, b) => a - b)
    let count = 0;
    let j = 0;
    
    for (let i = 0; i < g.length; i++) {
        while (j < s.length && s[j] < g[i]) {
            j++
        }
        
        if (j < s.length) {
            count++
            j++
        } else {
            break
        }
    }
    return count
};

File: 0455-assign-cookies/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        g.sort()
        s.sort()
        count = 0
        j = 0
        
        for i in range(len(g)):
            while j < len(s) and s[j] < g[i]:
                j += 1
            
            if j < len(s):
                count += 1
                j += 1
            else:
                break
        return count
        
